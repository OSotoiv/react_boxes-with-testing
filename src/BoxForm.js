import React from "react";

const BoxForm = ({ submitBoxForm, updateFormState, formState }) => {
    return (<>
        <form onSubmit={submitBoxForm}>
            <label htmlFor="width">Width:</label>
            <input onChange={updateFormState} value={formState.width} type="number" id="width" name="width" placeholder="Enter width" required />

            <label htmlFor="height">Height:</label>
            <input onChange={updateFormState} value={formState.height} type="number" id="height" name="height" placeholder="Enter height" required />

            <button type="submit">Submit</button>
        </form>
    </>)
}
export default BoxForm;