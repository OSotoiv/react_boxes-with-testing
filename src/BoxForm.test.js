import React from "react";
import { render, fireEvent } from "@testing-library/react";
import BoxForm from "./BoxForm";

it('renders SMOKE TEST', () => {
    render(<BoxForm formState={{ width: '', height: '' }} submitBoxForm={(e) => e.preventDefault()} updateFormState={(e) => e.preventDefault()} />);
});

it("matches snapshot", function () {
    const { asFragment } = render(<BoxForm formState={{ width: '', height: '' }} submitBoxForm={(e) => e.preventDefault()} updateFormState={(e) => e.preventDefault()} />);
    expect(asFragment()).toMatchSnapshot();
});

it('should have empty inputs for width and height', () => {
    const { getByLabelText } = render(<BoxForm formState={{ width: '', height: '' }} submitBoxForm={(e) => e.preventDefault()} updateFormState={(e) => e.preventDefault()} />);
    const widthInput = getByLabelText('Width:');
    const heightInput = getByLabelText('Height:');
    expect(widthInput).toBeInTheDocument();
    expect(widthInput).toHaveValue(null);
    expect(heightInput).toBeInTheDocument();
    expect(heightInput).toHaveValue(null);
});

it('should have a submit button', () => {
    const { getByText } = render(<BoxForm formState={{ width: '', height: '' }} submitBoxForm={(e) => e.preventDefault()} updateFormState={(e) => e.preventDefault()} />);
    const button = getByText('Submit');
    expect(button).toBeInTheDocument();
});