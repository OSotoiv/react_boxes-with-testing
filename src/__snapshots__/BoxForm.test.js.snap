// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <form>
    <label
      for="width"
    >
      Width:
    </label>
    <input
      id="width"
      name="width"
      placeholder="Enter width"
      required=""
      type="number"
      value=""
    />
    <label
      for="height"
    >
      Height:
    </label>
    <input
      id="height"
      name="height"
      placeholder="Enter height"
      required=""
      type="number"
      value=""
    />
    <button
      type="submit"
    >
      Submit
    </button>
  </form>
</DocumentFragment>
`;
